internal/cloud/aws/aws.go

```diff
@@ -126,46 +130,98 @@ func (c *Cloud) InitSecretHash(ctx context.Context) ([]byte, error) {
+		return *describeIPsOutput.Addresses[0].PublicIp, strconv.FormatInt(constants.KubernetesPort, 10), nil
 	}
-	if len(arns) != 1 {
-		return "", fmt.Errorf("%d load balancers found", len(arns))
+	// fallback to old architecture
+	// this will be removed in the future
+	hostname, err := c.getLoadBalancerIPOldInfrastructure(ctx)
+	if err != nil {
+		return "", "", fmt.Errorf("retrieving load balancer ip: %w", err)
 	}
+	return hostname, strconv.FormatInt(constants.KubernetesPort, 10), nil
+}
 
-	output, err := c.loadbalancer.DescribeLoadBalancers(ctx, &elasticloadbalancingv2.DescribeLoadBalancersInput{
-		LoadBalancerArns: arns,
-	})
+// getLoadBalancerIPOldInfrastructure returns the IP of the load balancer.
+// This is only used for the old infrastructure.
+// This will be removed in the future.
+func (c *Cloud) getLoadBalancerIPOldInfrastructure(ctx context.Context) (string, error) {
+	loadbalancer, err := c.getLoadBalancer(ctx)
 	if err != nil {
-		return "", fmt.Errorf("retrieving load balancer: %w", err)
-	}
-	if len(output.LoadBalancers) != 1 {
-		return "", fmt.Errorf("%d load balancers found; expected 1", len(output.LoadBalancers))
+		return "", fmt.Errorf("finding Constellation load balancer: %w", err)
 	}
 
 	// TODO(malt3): Add support for multiple availability zones in the lb frontend.
 	// This can only be done after we have migrated to using DNS as the load balancer endpoint.
 	// At that point, we don't need to care about the number of availability zones anymore.
-	if len(output.LoadBalancers[0].AvailabilityZones) != 1 {
-		return "", fmt.Errorf("%d availability zones found; expected 1", len(output.LoadBalancers[0].AvailabilityZones))
+	if len(loadbalancer.AvailabilityZones) != 1 {
+		return "", fmt.Errorf("%d availability zones found; expected 1", len(loadbalancer.AvailabilityZones))
 	}
 
-	if len(output.LoadBalancers[0].AvailabilityZones[0].LoadBalancerAddresses) != 1 {
-		return "", fmt.Errorf("%d load balancer addresses found; expected 1", len(output.LoadBalancers[0].AvailabilityZones[0].LoadBalancerAddresses))
+	if len(loadbalancer.AvailabilityZones[0].LoadBalancerAddresses) != 1 {
+		return "", fmt.Errorf("%d load balancer addresses found; expected 1", len(loadbalancer.AvailabilityZones[0].LoadBalancerAddresses))
 	}
-	if output.LoadBalancers[0].AvailabilityZones[0].LoadBalancerAddresses[0].IpAddress == nil {
+	if loadbalancer.AvailabilityZones[0].LoadBalancerAddresses[0].IpAddress == nil {
 		return "", errors.New("load balancer address is nil")
 	}
 
-	// TODO(malt3): ideally, we would use DNS here instead of IP addresses.
-	// Requires changes to the infrastructure.
-	return *output.LoadBalancers[0].AvailabilityZones[0].LoadBalancerAddresses[0].IpAddress, nil
+	return *loadbalancer.AvailabilityZones[0].LoadBalancerAddresses[0].IpAddress, nil
+}
+
+// TODO(malt3): uncomment and use as soon as we switch the primary endpoint to DNS.
+// func (c *Cloud) getLoadBalancerDNSName(ctx context.Context) (string, error) {
```

Comment:
I'd favor commenting this out via multi-line comments `/**/`. This way the latest `git blame` for those lines stays with the latest changes to the code itself.